### ISSUE TITLE
[Apple Notes] All attachments become "**error reading attachment**" after import

### ISSUE THREAD

**Issue by dinghaoz** [obsidianmd/obsidian-importer#148]
I am on macOS 14.0 (23A344). Zero attachments are actually imported.

**Comment by mirnovov** [comment]
@dinghaoz This is because the importer is having trouble getting the path to where your attachments are stored. Unfortunately

- Do you use 'On My Mac' instead of storing your Apple Notes in iCloud?
- Are any of the notes you've trying to import shared with other people, or by other people?
- Your files should be stored in `/Users/yourusername/Library/Group Containers/group.com.apple.notes/Accounts/bunchofrandomnumbers/Media/`. Are there images in the subfolders there?
- Does your user folder have any special characters or is otherwise unusual?

**Comment by dinghaoz** [comment]
Thanks for the reply. Here are the answers:
1. I store my Apple Notes in iCloud.
2. Yes, I do share a few notes with my wife, but not all.
3. No, /Users/yourusername/Library/Group Containers/group.com.apple.notes/Accounts/bunchofrandomnumbers/ doesn't have Media folder. But /Users/yourusername/Library/Group Containers/group.com.apple.notes/Media exists and it seems like it is where all the attachments located.
4. Yes, they do.
5. Yes, I prefix my notes folder with emojis to make them look more like Notion and my notes' titles are mostly in Chinese.
6. Yes I have.

**Comment by dinghaoz** [comment]
I feel that the issue might be related to question 3. The code seems assuming the attachments are under `group.com.apple.notes/Accounts/account-UUID/Media` but they are actually at `group.com.apple.notes`.

```javascript
				const account = row.ZACCOUNT6 || row.ZACCOUNT5;
				sourcePath = path.join(
					this.resolvedAccounts[account].path, 
					'Media', row.ZIDENTIFIER, row.ZGENERATION1 || '', row.ZFILENAME
				);
```

But my folder hierarchy looks like this
<img width="358" alt="Screenshot 2023-10-21 at 23 27 00" src="https://github.com/obsidianmd/obsidian-importer/assets/26001134/cde828e8-b03e-421e-b3ef-0658384dd592">

**Comment by mirnovov** [comment]
Yeah, I think this is it.

It’s not to do with the macOS version I think, I’m on Sonoma and everything is under the Accounts folder for me. Worst case scenario if I can’t figure out I can always just have it check both the first time to determine which one to use.

**Comment by kepano** [comment]
@mirnovov there may be some useful prior art in [Apple Notes Liberator](https://github.com/HamburgChimps/apple-notes-liberatoro) to find the location of the database

**Comment by mirnovov** [comment]
Unfortunately there isn’t, it assumes the notes are in the Accounts folder like this importer currently does.

**Comment by kepano** [comment]
This should now be fixed if you update to Importer 1.5.3
